flowzone.yml

```diff
@@ -1529,87 +1535,39 @@ jobs:
             echo "pypi_publish=false" >> $GITHUB_OUTPUT
           fi
 
-      # Check which Python 3.7+ versions meet the Poetry project requirements
-      - name: Install Poetry
+      - name: Install Poetry tool
         if: steps.python_poetry.outputs.enabled == 'true'
         run: |
           pipx install poetry
 
-      - name: Set up Python 3.7
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.7"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.7 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.7"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.7 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.8
+      - name: Set up Python environment
         if: steps.python_poetry.outputs.enabled == 'true'
         uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
         with:
-          python-version: "3.8"
+          python-version: ${{ env.SUPPORTED_VERSIONS }}
```

Comment:
Multiple versions supported by the action, see https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-multiple-pythonpypy-version